### PR TITLE
[INFRA] Intel compiler support

### PR DESCRIPTION
**Pull request by eseiler**
Blocked by https://github.com/seqan/seqan/pull/2448

The last commits contains some fixes (workarounds) that make icpc work for me.
Also see https://github.com/seqan/product_backlog/issues/261#issuecomment-748262888

### CODE REVIEW

**Comment by eseiler** [comment]
!rebuild

**Comment by eseiler** [comment]
!rebuild

**Comment by eseiler** [comment]
We still need all the changes with `icc (ICC) 2021.1.2 20201208`

File: include/seqan/align/dp_cell_dynamic.h [hunk 1]
```diff
@@ -184,10 +184,12 @@ inline SEQAN_FUNC_ENABLE_IF(Not<Is<SimdVectorConcept<TScoreValue> > >,bool)
 isGapExtension(DPCell_<TScoreValue, DynamicGaps> const & cell,
                TSpec const & /*spec*/)
 {
+    bool result;
     SEQAN_IF_CONSTEXPR (IsSameType<TSpec, DynamicGapExtensionHorizontal>::VALUE)
-        return cell._flagMask & MASK_HORIZONTAL_GAP;
+        result =cell._flagMask & MASK_HORIZONTAL_GAP;
```
Comment (eseiler) [inline include/seqan/align/dp_cell_dynamic.h:189]:
```suggestion
        result = cell._flagMask & MASK_HORIZONTAL_GAP;
```

**Review by marehr — APPROVED** [review]

**Comment by eseiler** [comment]
@marehr Can we merge this? Then the next step would be to set up the _iccci_ :)

**Pull request merged**